### PR TITLE
fix name can not be reset after node is destroyed

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/core/assets/Asset.cpp [hunk 1]
```diff
@@ -101,4 +101,10 @@ bool Asset::destroy() {
     return Super::destroy();
 }
 
+void Asset::destruct() {
+    CCObject::destruct();
+    _native = "";
+    _nativeUrl = "";
```
Comment (dumganhar) [inline native/cocos/core/assets/Asset.cpp]: Use `_native.clear()`, `_nativeUrl.clear()`

File: cocos/core/data/object.ts [hunk 2]
```diff
@@ -384,6 +384,9 @@ class CCObject implements EditorExtendableObject {
         }
 
         if (!EDITOR || legacyCC.GAME_VIEW) {
+            if (JSB &&  typeof (this.destruct) === 'function') {
```
Comment (dumganhar) [inline cocos/core/data/object.ts]: Why need to check whether it's a function? Is it possible not a function?

File: native/cocos/bindings/auto/jsb_ar_auto.cpp [hunk 3]
```diff
@@ -601,6 +601,49 @@ static bool js_cc_ar_ARModule_getCameraDepthBuffer(se::State& s)
...
+    size_t argc = args.size();
+    cc::ar::ARModule *arg1 = (cc::ar::ARModule *) NULL ;
+    bool result;
+    
+    if(argc != 0) {
+        SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 0);
+        return false;
+    }
+    arg1 = SE_THIS_OBJECT<cc::ar::ARModule>(s);
+    if (nullptr == arg1) return true;
```
Comment (dumganhar) [inline native/cocos/bindings/auto/jsb_ar_auto.cpp]: Why change this `auto generated` file? This PR doesn't make any change of swig configurations.

File: native/cocos/core/data/Object.cpp [hunk 4]
```diff
@@ -59,6 +59,10 @@ CCObject::CCObject(ccstd::string name /* = ""*/)
 
 CCObject::~CCObject() = default;
 
+void CCObject::destruct() {
+    _name = "";
```
Comment (dumganhar) [inline native/cocos/core/data/Object.cpp]: _name.clear()

File: cocos/core/data/object.ts (hunk 2 above)
Comment (bofeng-song) [reply cocos/core/data/object.ts]: Currently, it`s impossible, here we can check whether it is defined

File: native/cocos/bindings/auto/jsb_ar_auto.cpp (hunk 3 above)
Comment (bofeng-song) [reply native/cocos/bindings/auto/jsb_ar_auto.cpp]: I just make the action ' node generate.js '. Maybe someone forget to make the action

File: cocos/core/data/object.ts (hunk 2 above)
Comment (minggo) [inline cocos/core/data/object.ts]: Please add comment.

File: native/cocos/bindings/auto/jsb_ar_auto.cpp (hunk 3 above)
Comment (dumganhar) [reply native/cocos/bindings/auto/jsb_ar_auto.cpp]: pull the latest code and execute `node genbindings.js` again.

Comment (bofeng-song) [reply native/cocos/bindings/auto/jsb_ar_auto.cpp]: I had pull the latest code

File: native/cocos/core/data/Object.cpp [hunk 5]
```diff
@@ -59,6 +59,10 @@ CCObject::CCObject(ccstd::string name /* = ""*/)
 
 CCObject::~CCObject() = default;
 
+void CCObject::destruct() {
+    _name.clear();
```
Comment (dumganhar) [inline native/cocos/core/data/Object.cpp:63]: Does _objFlags also need to be reset to 0?
I think `destruct` method should reset all member variables in class.

Comment (bofeng-song) [reply native/cocos/core/data/Object.cpp:63]: The ts layer logic only resets string, object and function

Comment (bofeng-song) [reply native/cocos/core/data/Object.cpp:63]: If we reset all variables, the destroyed flag may be reset to initial state ?

Comment (dumganhar) [reply native/cocos/core/data/Object.cpp:63]: All right, _objFlags is an exception.

File: cocos/core/data/object.ts [hunk 6]
```diff
@@ -384,6 +384,16 @@ class CCObject implements EditorExtendableObject {
         }
 
         if (!EDITOR || legacyCC.GAME_VIEW) {
+            /*Native properties cannot be reset by _destruct, because the native properties are hung on the prototype and
+             *hasOwnProperty's detection cannot be passed.
+             */
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-expect-error
+            if (JSB &&  this.destruct) {
```
Comment (dumganhar) [inline cocos/core/data/object.ts]: an extra space after &&

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**